Src/FluentAssertions.Json/JTokenDifferentiator.cs

```diff
@@ -1,13 +1,15 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using FluentAssertions.Equivalency;
 using Newtonsoft.Json.Linq;
 
 namespace FluentAssertions.Json
 {
     internal static class JTokenDifferentiator
     {
-        public static Difference FindFirstDifference(JToken actual, JToken expected, bool ignoreExtraProperties)
+
+        public static Difference FindFirstDifference(JToken actual, JToken expected, bool ignoreExtraProperties, Func<EquivalencyAssertionOptions<object>, EquivalencyAssertionOptions<object>> config)
```

Comment:
🤔 Looking at the number of methods that are affected by that new `config` paramater, it feels like it is the time to refactor this class to become an instance class. Then you can make parameters like `path`, `ignoreExtraProperties` and `config` instance fields. However, I fully understand if you don't want to do that now. In that case, I recommend adding a comment above the class definition like:

`// REFACTOR: Change to non-static and make parameters fields`